### PR TITLE
[179] Notification banner shown on jobseeker account on sign in after feature release

### PR DESCRIPTION
**Pull request by georges1996**
https://trello.com/c/7aCsfF4e/179-notification-banner-shown-on-jobseeker-account-on-first-sign-in-after-feature-release

## Changes in this PR:

- Show the profiles banner when their profile is incomplete

### CODE REVIEW

**Comment by bencmitchell** [comment]
[Notification banner shown on jobseeker account on first sign in after feature release](https://trello.com/c/7aCsfF4e/179-notification-banner-shown-on-jobseeker-account-on-first-sign-in-after-feature-release)

File: app/controllers/jobseekers/saved_jobs_controller.rb [hunk 1]
```diff
@@ -2,6 +2,8 @@ class Jobseekers::SavedJobsController < Jobseekers::BaseController
   helper_method :saved_jobs, :sort
 
   def index
+    @show_candidate_profiles_banner = show_candidate_profiles_banner
+
```
Comment (elliotcm) [inline app/controllers/jobseekers/saved_jobs_controller.rb:6]:
```suggestion
```

File: app/controllers/jobseekers/saved_jobs_controller.rb [hunk 2]
```diff
@@ -22,6 +24,15 @@ def destroy
...
   private
 
+  def show_candidate_profiles_banner
+    @profile ||= JobseekerProfile.where(jobseeker_id: current_jobseeker.id)
+
+    personal_details = PersonalDetails.where(jobseeker_profile_id: @profile).first
+    job_preferences = JobPreferences.where(jobseeker_profile_id: @profile).first
+
+    section_completed(personal_details) && section_completed(job_preferences)
+  end
```
Comment (elliotcm) [inline app/controllers/jobseekers/saved_jobs_controller.rb:34]:
```suggestion
  def show_candidate_profiles_banner?
    profile = current_jobseeker.jobseeker_profile
    personal_details = profile.personal_details
    job_preferences = profile.job_preferences

    section_completed(personal_details) && section_completed(job_preferences)
  end
  helper_method :show_candidate_profiles_banner?
```

File: app/controllers/jobseekers/saved_jobs_controller.rb [hunk 3]
```diff
@@ -41,4 +52,11 @@ def sort
   def vacancy
     @vacancy ||= Vacancy.listed.find(saved_job_params[:job_id])
   end
+
+  def section_completed(object)
+    return unless object && object["completed_steps"].any?
+
+    step_phases = %w[completed skipped]
+    object["completed_steps"].values.all? { |step| step_phases.include?(step) }
+  end
```
Comment (elliotcm) [inline app/controllers/jobseekers/saved_jobs_controller.rb:61]:
```suggestion
  def section_completed(record)
    return if record.completed_steps.blank?

    step_phases = %w[completed skipped]
    record.completed_steps.values.all? { |step| step_phases.include?(step) }
  end
```

File: app/views/jobseekers/saved_jobs/index.html.slim [hunk 4]
```diff
@@ -13,6 +13,8 @@ h1.govuk-heading-l = t(".page_title")
 
 .govuk-grid-row
   .govuk-grid-column-full
+    - unless @show_candidate_profiles_banner
+      = render "candidiate_profiles_banner"
```
Comment (elliotcm) [inline app/views/jobseekers/saved_jobs/index.html.slim:17]: This seems to be inverted?

```suggestion
    = render "candidiate_profiles_banner" if show_candidate_profiles_banner?
```

**Review by elliotcm — CHANGES_REQUESTED** [review]

**Comment by bencmitchell** [comment]
@georgeschena looks good to me 👍

**Review by elliotcm — APPROVED** [review]

**Pull request merged**